macros/src/codegen.rs

```diff
@@ -224,105 +252,107 @@ pub fn generate_code(sm: &ParsedStateMachine) -> proc_macro2::TokenStream {
 
     let mut guard_list = proc_macro2::TokenStream::new();
     let mut action_list = proc_macro2::TokenStream::new();
-    for (state, value) in transitions.iter() {
+    for (state, event_mappings) in transitions.iter() {
         // create the state data token stream
         let state_data = match sm.state_data.data_types.get(state) {
             Some(st @ Type::Reference(_)) => quote! { state_data: #st, },
             Some(st) => quote! { state_data: &#st, },
             None => quote! {},
         };
 
-        value.iter().for_each(|(event, value)| {
-            // get input state lifetimes
-            let in_state_lifetimes = sm.state_data.lifetimes.get(&value.in_state.to_string()).cloned().unwrap_or_default();
-
-            // get output state lifetimes
-            let out_state_lifetimes = sm.state_data.lifetimes.get(&value.out_state.to_string()).cloned().unwrap_or_default();
-
-            // get event lifetimes
-            let event_lifetimes = sm.event_data.lifetimes.get(event).cloned().unwrap_or_default();
-
-            // combine all lifetimes
-            let mut all_lifetimes = Lifetimes::new();
-            all_lifetimes.extend(&in_state_lifetimes);
-            all_lifetimes.extend(&out_state_lifetimes);
-            all_lifetimes.extend(&event_lifetimes);
-
-            // Create the guard traits for user implementation
-            if let Some(AsyncIdent {ident: guard, is_async}) = &value.guard {
-                let event_data = match sm.event_data.data_types.get(event) {
-                    Some(et @ Type::Reference(_)) => quote! { event_data: #et },
-                    Some(et) => quote! { event_data: &#et },
-                    None => quote! {},
-                };
-
-                let guard_error = if sm.custom_guard_error {
-                    quote! { Self::GuardError }
-                } else {
-                    quote! { () }
-                };
-
-                // Only add the guard if it hasn't been added before
-                if !guard_set.iter().any(|g| g == guard) {
-                    guard_set.push(guard.clone());
+        event_mappings.iter().for_each(|(event, event_mapping)| {
+            event_mapping.transitions.iter().for_each(|transition|{
```

Comment:
Can we just unwrap these to normal for loops? I don't see the purpose in using `for_each` on an iterator here.

Reply:
This is a more idiomatic rust (functional style) which is the approach advocated by the rust book. 
If I have right to choose I'd prefer this style :)